diff -up libobjc2-2.1/CMakeLists.txt.omv~ libobjc2-2.1/CMakeLists.txt
--- libobjc2-2.1/CMakeLists.txt.omv~	2021-04-12 18:47:37.350451601 +0200
+++ libobjc2-2.1/CMakeLists.txt	2021-04-12 18:58:28.350863061 +0200
@@ -316,9 +316,10 @@ if (ENABLE_OBJCXX)
 				set(ENABLE_OBJCXX false)
 			endif()
 		endif ()
+		separate_arguments(SPLITCXXFLAGS UNIX_COMMAND "${CMAKE_CXX_FLAGS}")
 		add_custom_command(OUTPUT eh_trampoline.s
-			COMMAND ${CMAKE_CXX_COMPILER} ${CMAKE_CXX_FLAGS} -fPIC -S "${CMAKE_SOURCE_DIR}/eh_trampoline.cc" -o - -fexceptions -fno-inline | sed "s/__gxx_personality_v0/test_eh_personality/g" > "${CMAKE_BINARY_DIR}/eh_trampoline.s"
-			MAIN_DEPENDENCY eh_trampoline.cc)
+			COMMAND ${CMAKE_CXX_COMPILER} ${SPLITCXXFLAGS} -fPIC -S "${CMAKE_SOURCE_DIR}/eh_trampoline.cc" -o - -fexceptions -fno-inline -fno-lto | sed "s/__gxx_personality_v0/test_eh_personality/g" > "${CMAKE_BINARY_DIR}/eh_trampoline.s"
+			MAIN_DEPENDENCY eh_trampoline.cc VERBATIM)
 		list(APPEND libobjc_ASM_SRCS eh_trampoline.s)
 		list(APPEND libobjc_CXX_SRCS objcxx_eh.cc)
 	endif ()
